node/pkg/watchers/hedera/watcher.go

```diff
@@ -0,0 +1,612 @@
+	logDataString = logDataString[2:] // remove the leading 0x
+	logDataBytes, err := hex.DecodeString(logDataString)
+	if err != nil {
+		logger.Fatal("failed to unpack log data", zap.Error(err))
+		return nil, fmt.Errorf("Data field could not be converted for txhash %s, value[%s]", txhash)
+	}
+
+	unpackedMsg, err := ethAbi.Unpack("LogMessagePublished", logDataBytes)
+	if err != nil {
+		logger.Fatal("failed to unpack log data", zap.Error(err))
+		return nil, fmt.Errorf("Log Data field could not be unpacked for txhash %s", txhash)
+	}
+
+	// // AbiLogMessagePublished represents a LogMessagePublished event raised by the Abi contract.
+	// type AbiLogMessagePublished struct {
+	// 	Sender           common.Address
+	// 	Sequence         uint64
+	// 	Nonce            uint32
+	// 	Payload          []byte
+	// 	ConsistencyLevel uint8
+	// 	Raw              types.Log // Blockchain specific contextual infos
+	// }
+
+	seq := unpackedMsg[0].(uint64)
+	nonce := unpackedMsg[1].(uint32)
+	payload := unpackedMsg[2].([]byte)
+	cLevel := unpackedMsg[3].(uint8)
+	logger.Info("unpackedMsg",
+		zap.Int("Length of unpackedMsg", len(unpackedMsg)),
+		zap.Uint64("Sequence", seq),
+		zap.Uint32("Nonce", nonce),
+		zap.Uint8("consistencyLevel", cLevel))
+	var chainID vaa.ChainID = 27
+
+	// Convert timestamp
+	timeVals := strings.Split(timeStampBase.String(), ".")
+	timeValSeconds, err := strconv.ParseInt(timeVals[0], 10, 64)
+	if err != nil {
+		logger.Error("Failed to convert seconds", zap.String("sec", timeVals[0]))
+		return nil, fmt.Errorf("Failed to convert time seconds %s", timeVals[0])
+	}
+	timeValNS, err := strconv.ParseInt(timeVals[1], 10, 64)
+	if err != nil {
+		logger.Error("Failed to convert nanoseconds", zap.String("nanosec", timeVals[1]))
+		return nil, fmt.Errorf("Failed to convert time nanoseconds %s", timeVals[1])
+	}
+	unixTS := time.Unix(timeValSeconds, timeValNS)
+	logger.Info("timestamps", zap.Stringer("base", timeStampBase), zap.String("slice0", timeVals[0]), zap.String("slice1", timeVals[1]), zap.Stringer("unix", unixTS))
+
+	messagePublication := &common.MessagePublication{
+		TxHash:           txHash,      // In log
+		Timestamp:        unixTS,      // In log
+		Nonce:            nonce,       // In log data
+		Sequence:         seq,         // In log data
+		EmitterChain:     chainID,     // Don't know where to get this
+		EmitterAddress:   emitterAddr, // In log
+		Payload:          payload,     // In log data
+		ConsistencyLevel: cLevel,      // In log data
+	}
+
+	logger.Info("messagePublication",
+		zap.Stringer("txHash", txHash),
+		zap.Stringer("timestamp", unixTS),
+		zap.Uint32("nonce", nonce),
+		zap.Uint64("sequence", seq),
+		zap.Stringer("EmitterAddr", emitterAddr),
+		zap.Uint8("ConsistenceLevel", cLevel),
+	)
+
+	return messagePublication, nil
+}
+
+// TxMessageToEvent a JSON structured "contracts" output and converts it into an event
+func TxMessageToEvent(logger *zap.Logger, msg string) ([]*common.MessagePublication, error) {
```

Comment:
Pass in chain ID so you can use it below? Or remove it from the watcher object and just hard code it to vaa.ChainIDHedera, since this is the Hedera watcher.

Reply:
Fixed.